modules/gapi/include/opencv2/gapi/gkernel.hpp

```diff
@@ -233,29 +234,28 @@ class GKernelType<K, std::function<R(Args...)> >
 #define G_ID_HELPER_CLASS(Class)  Class##IdHelper
 
 #define G_ID_HELPER_BODY(Class, Id)                                         \
-    namespace detail                                                        \
+    struct G_ID_HELPER_CLASS(Class)                                         \
     {                                                                       \
-        struct G_ID_HELPER_CLASS(Class)                                     \
-        {                                                                   \
-            static constexpr const char * id() {return Id;}                 \
-        };                                                                  \
-    }
+        static constexpr const char * id() {return Id;}                     \
+    };                                                                      \
 
 #define G_TYPED_KERNEL(Class, API, Id)                                      \
     G_ID_HELPER_BODY(Class, Id)                                             \
     struct Class final: public cv::GKernelType<Class, std::function API >,  \
-                        public detail::G_ID_HELPER_CLASS(Class)
-// {body} is to be defined by user
+                        public G_ID_HELPER_CLASS(Class)
+    // {body} is to be defined by user
 
 #define G_TYPED_KERNEL_M(Class, API, Id)                                    \
     G_ID_HELPER_BODY(Class, Id)                                             \
     struct Class final: public cv::GKernelTypeM<Class, std::function API >, \
-                        public detail::G_ID_HELPER_CLASS(Class)
-// {body} is to be defined by user
+                        public G_ID_HELPER_CLASS(Class)
+    // {body} is to be defined by user
 
 #define G_API_OP   G_TYPED_KERNEL
 #define G_API_OP_M G_TYPED_KERNEL_M
 
+//! @endcond
```

Comment:
G_TYPED_KERNEL and other macros shouldn't be ingored in the Doxygen, it is part of public API

Reply:
Applied.